dissect/target/plugins/os/windows/_os.py

```diff
@@ -314,6 +314,25 @@ def users(self) -> Iterator[WindowsUserRecord]:
                     _target=self.target,
                 )
 
+    @internal
+    def misc_user_paths(self) -> Iterator[tuple[str, tuple[str, str] | None]]:
+        yield from (
+            (resolved_path, user_criterion)
+            for path, user_criterion in [
+                ("%windir%/ServiceProfiles/LocalService", ("sid", "S-1-5-19")),
+                ("%windir%/ServiceProfiles/NetworkService", ("sid", "S-1-5-20")),
+                ("%windir%/System32/config/systemprofile", ("sid", "S-1-5-18")),
+                ("%windir%/SysWOW64/config/systemprofile", ("sid", "S-1-5-18")),
+            ]
+            if (resolved_path := self.target.resolve(path)).exists()
+        )
+
+        # Homedirs for users when there is no user profile information available.
+        user_dirs = ("sysvol/Users", "sysvol/Documents and Settings")
+        for user_dir in user_dirs:
+            if (resolved := self.target.resolve(user_dir)).exists():
+                yield from ((entry, None) for entry in resolved.iterdir() if entry.is_dir())
```

Comment:
Why not make this a list comprehension too like the one above? Or at least put the `user_dirs` tuple inside the loop?

Reply:
> Why not make this a list comprehension too like the one above? 

With multiple nestings / sources I find the python list comprehension to be less clear than an explicit loop

> Or at least put the user_dirs tuple inside the loop?

Sure